openproblems/tasks/_batch_integration/batch_integration_feature/api.py

```diff
@@ -1,15 +1,10 @@
-from ....data.sample import load_sample_data
-from ....tools.decorators import dataset
-
-import numpy as np
+from .._common import api
 
 
 def check_dataset(adata):
     """Check that dataset output fits expected API."""
+    api.check_dataset(adata)
 
-    assert "batch" in adata.obs
-    assert "labels" in adata.obs
-    assert "log_normalized" in adata.layers
     assert "counts" in adata.layers
     assert adata.var_names.is_unique
     assert adata.obs_names.is_unique
```

Comment:
counts, var_names, and obs_names check can (and probably should) also be put into the common api.